[Kineto][NCCL][5/n] Populate in/out split size info for all_to_all from CPU to CUDA kernel

yoyoyocmu opened the pull request:
Summary: This diff populates all_to_all input and out split size from CPU op to GPU kernel when valid.

Test Plan:
**Trace example**:
- For non all_to_all collective functions: https://fburl.com/perfdoctor/4nobsu15
https://pxl.cl/3GNVb

- For all_to_all: https://fburl.com/perfdoctor/f418goys

https://pxl.cl/3H2nd

Differential Revision: D50762093

aaronenyeshi commented on torch/csrc/profiler/util.cpp:
```diff
@@ -365,12 +366,18 @@ std::unordered_map<std::string, std::string> saveNcclMeta(
   map.emplace(kOutMsgSize, std::to_string(debugInfo->getOutMessageSize()));
-  map.emplace(
+  auto& inSplitSizes = debugInfo->getInputSplitSizes();
+  if (!inSplitSizes.empty() && inSplitSizes.size() <= kTruncatLength) {
```
Should we still record the first kTruncatLength in the list?

yoyoyocmu replied:
Good point, i was thinking should we keep the first 1 or 30 elements in the list or just skip recording if the length is too long. Any recommended rule to follow?

aaronenyeshi replied:
It would make sense to record the first 30 elements, and then add a ... when the length is greater than 30. So then we can claim to show all elements up to the first 30.

aaronenyeshi replied:
We can address this in the future, if someone is interested.

aaronenyeshi reviewed (approved):
LGTM!

yoyoyocmu committed:
[Kineto][NCCL][5/n] Populate in/out split size info for all_to_all from CPU to CUDA kernel (#112308)

Summary:

X-link: https://github.com/pytorch/kineto/pull/822

This diff populates all_to_all input and out split size from CPU op to GPU kernel when valid.

Test Plan:
**Trace example**:
- For non all_to_all collective functions: https://fburl.com/perfdoctor/4nobsu15
https://pxl.cl/3GNVb

- For all_to_all: https://fburl.com/perfdoctor/f418goys

https://pxl.cl/3H2nd

Reviewed By: aaronenyeshi, idning

Differential Revision: D50762093
